### PR TITLE
[JFG-1091] How to set limits for percentiles

### PR DESCRIPTION
**Pull request by SokolAndrey**
Shall I rename LATENCY_AVG to LATENCY?

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java [hunk 1]
```diff
@@ -53,7 +53,26 @@
     MON_NONHEAP_MAX,
     MON_THREAD_COUNT,
     MON_THREAD_PEAK_COUNT,
-    MON_FILE_DESCRIPTORS;
+    MON_FILE_DESCRIPTORS,
+    LATENCY_AVG,
+    LATENCY_STD_DEV;
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: we have already PERF_LATENCY_AVG and PERF_LATENCY_STD_DEV

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java [hunk 2]
```diff
@@ -53,7 +53,26 @@
...
+    private final double DEFAULT_VALUE = -1D;
+
+    private double value = DEFAULT_VALUE;
+
+    /**
+     * Metric name for some particular percentile.
+     *
+     * @param percentile the value of percentile. It may varies from 0 to 100.0 percent.
+     */
+    public static JMetricName LATENCY_PERCENTILE(Double percentile) {
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: please add PERF_ prefix

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java [hunk 3]
```diff
@@ -204,6 +223,16 @@ public String transformToString() {
                 break;
             case MON_FILE_DESCRIPTORS:
                 name = StandardMetricsNamesUtil.MON_FILE_DESCRIPTORS;
+                break;
+            case LATENCY_STD_DEV:
+                name = StandardMetricsNamesUtil.LATENCY_STD_DEV_ID;
+                break;
+            case LATENCY_AVG:
+                if (this.value == DEFAULT_VALUE) {
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: you can't use == for doubles

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java [hunk 4]
```diff
@@ -204,6 +223,16 @@ public String transformToString() {
...
+                    name = StandardMetricsNamesUtil.LATENCY_ID;
+                } else {
+                    name = StandardMetricsNamesUtil.LATENCY + " " + this.value + " %";
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: StandardMetricsNamesUtil .getLatencyMetricName()

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java [hunk 5]
```diff
@@ -53,7 +53,26 @@
     MON_NONHEAP_MAX,
     MON_THREAD_COUNT,
     MON_THREAD_PEAK_COUNT,
-    MON_FILE_DESCRIPTORS;
+    MON_FILE_DESCRIPTORS,
+    LATENCY_AVG,
+    LATENCY_STD_DEV;
+
+    private final double DEFAULT_VALUE = -1D;
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: please, comment why these variables are needed

Comment (SokolAndrey) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: Cause we want to allow users set any value for percentile.

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java (hunk 3 above)
Comment (SokolAndrey) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: It is posible here

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java (hunk 4 above)
Comment (SokolAndrey) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/limits/auxiliary/JMetricName.java]: Clarify pls

File: archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java [hunk 6]
```diff
@@ -103,8 +113,9 @@ public JLoadScenario exampleJaggerLoadScenario() {
 
         // For JLoadScenario which is supposed to be executed by Jagger its ID must be set to 'jagger.load.scenario.id.to.execute' property's value
         return JLoadScenario.builder(Id.of("exampleJaggerLoadScenario"), jParallelTestsGroup)
-                            .addListener(new ExampleLoadScenarioListener())
```
Comment (anton-antonenko) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: Why did you remove it?

Comment (SokolAndrey) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: It wasn't me. Git has decided that I removed it. I just added percentiles. :ok_man:
